src/middlewared/middlewared/etc_files/local/minio/configure.py

```diff
@@ -41,3 +40,15 @@ def render(service, middleware):
             f.write(cert['privatekey'])
         os.chown(minio_privatekey, minio_uid, minio_gid)
         os.chmod(minio_privatekey, 0o600)
+
+
+def configure_minio_sys_dir(s3):
+    storage_path = s3['storage_path']
+    minio_dir = os.path.join(storage_path, '.minio.sys')
+    shutil.rmtree(minio_dir, ignore_errors=True)
```

Comment:
So we remove that dir everytime? Cant that cause trouble?

Reply:
Nope. Minio recognises buckets by the directories and does not save them in the `.minio.sys` configuration directory.